src/pages/api/searchOperators.ts

```diff
@@ -0,0 +1,39 @@
+import { NextApiResponse } from 'next';
+import { ErrorInfo, NextApiRequestWithSession } from '../../interfaces';
+import { SEARCH_OPERATOR_ATTRIBUTE } from '../../constants';
+import { redirectTo, redirectToError } from './apiUtils';
+import { updateSessionAttribute } from '../../utils/sessions';
+
+export interface SearchOperators {
+    searchResults: string[];
+}
+
+export interface SearchOperatorsWithErrors extends SearchOperators {
+    errors: ErrorInfo[];
+}
+
+export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
+    const errors: ErrorInfo[] = [];
+
+    try {
+        const { searchText } = req.body;
+
+        if (searchText.length < 4) {
```

Comment:
If this is the only error check we do, could we combine the 2 ifs into 
```
if (searchText.length < 4) {
const errors = [{
                errorMessage: 'Search requires a minimum of three characters',
                id: 'searchText',
                userInput: searchText,
            }]
updateSessionAttribute(req, SEARCH_OPERATOR_ATTRIBUTE, { errors, searchResults: [] });
            redirectTo(res, '/searchOperators');
        }
```
And this way we can render the search they made on the page?

Reply:
i did it so in the next part of the work if we need to do more error handling like at least one checkbox checked etc.

So might be worth leaving it in?